src/next.ts

```diff
@@ -1,70 +1,117 @@
 import type { NextApiRequest, NextApiResponse } from "next";
+import { NextRequest } from "next/server";
 import {
   InngestCommHandler,
   ServeHandler,
 } from "./components/InngestCommHandler";
 import { headerKeys, queryKeys } from "./helpers/consts";
-import { processEnv } from "./helpers/env";
+import { isEdgeRuntime, processEnv } from "./helpers/env";
+import { RegisterOptions } from "./types";
 
 export const name = "nextjs";
 
+/**
+ * Assert whether the given request is a Next.js API request or an `NextRequest`
+ * from an edge function.
+ */
+const isEdgeRequest = (
+  req: NextApiRequest | NextRequest
+): req is NextRequest => {
+  return typeof req?.headers?.get === "function";
```

Comment:
This is in `src/next` which is almost always deployed to vercel, but should this be platform-specific instead of framework specific?

Reply:
@tonyhb Have made these changes - see https://github.com/inngest/inngest-js/pull/163/files#diff-5087958ec6f8b087e8827c51631a00210094a1d599c750e7ccade1b12ffc59c5R536-R542